Services/Badge/classes/class.ilBadgeHandler.php

```diff
@@ -541,14 +545,14 @@ public function sendNotification(
                     $url = new ilNotificationLink(new ilNotificationParameter('badge_notification_badges_goto', [], 'badge'), $url);
                     $osd_params = ["badge_list" => implode(", ", $user_badges)];
 
-                    $notification = new ilNotificationConfig(BadgeNotificationProvider::NOTIFICATION_TYPE);
+                    $notification = $provider->getNotificationConfig();
```

Comment:
What is unclear to me is the coupling to the notification provider here. Conceptually we know, that some (**not**all) of the `Toast` elements collected and handled by the `GS` are based on the `OSD` notification channel. So depending on the concrete `Toast Provider`, there might (**not** must) be a dependency to the `OSD` notifications from the direction of the `Toast` provider. But IMO there should be no coupling to the `GS Notifications` here.